File: backend/schemas.py
from pydantic import BaseModel
from typing import List

class ScheduleItem(BaseModel):
    time: str
    task_id: int

class TaskItem(BaseModel):
    id: int
    name: str
    completed: bool = False

class DailyScheduleFormat(BaseModel):
    day: str
    tasks: list[TaskItem]
    schedule: list[ScheduleItem]        # must create ScheduleItem object since dict is not supported by OpenAI

class DraftRequest(BaseModel):
    tasks: List[str]
    day: str

class DraftResponse(BaseModel):
    schedule: DailyScheduleFormat

class ConfirmRequest(BaseModel):
    schedule: DailyScheduleFormat

class CompleteTasksRequest(BaseModel):
    day: str
    completed_task_ids: List[int]   # marks the task complete given task.ids

class ReflectionResponse(BaseModel):
    summary: str

File: frontend/src/types/schemas.ts
export interface TaskItem {
    id: number;
    name: string;
    completed: boolean;
}

export interface ScheduleItem {
    time: string;
    task_id: number;
}

export interface DailyScheduleFormat {
    day: string;
    tasks: TaskItem[];
    schedule: ScheduleItem[];
}

export interface CompleteTasksRequest {
    day: string;
    completed_task_ids: number[];
}

export interface ReflectionResponse {
    summary: string;
}
